README.md

```diff
@@ -1644,6 +1645,60 @@ variables](https://www.terraform.io/docs/configuration/environment-variables.htm
 prefix `TF_VAR_`, so one way to share a variable named `foo` between Terraform and Terragrunt is to set its value
 as the environment variable `TF_VAR_foo` and to read that value in using this `get_env()` built-in function.
 
+#### get_output
+
+`get_output(TARGET_TERRAGRUNT_CONFIG, [NAME])` returns the value of the output of the terraform module configured with
+the terragrunt config specified at `TARGET_TERRAGRUNT_CONFIG`. If provided a `NAME`, this function will return just the
+output indexed by `NAME`. Otherwise, this will return the all outputs as a map. The return type is preserved. For
+example, if the output is of type `number`, the returned value of `get_output` will also be of type `number`.
+
+Example:
+
+Suppose you had the following terragrunt live folder structure:
+
+```
+/terraform-code
+├── app1
+│   └── terragrunt.hcl
+├── tests
+│   ├── app2
+│   |   └── terragrunt.hcl
+│   └── app3
+│       └── terragrunt.hcl
+```
+
+Suppose `app1` had the following outputs:
+
+```
+output "instance_id" {}
+output "instance_name" {}
+```
+
+In `app3`, you can get each of these outputs using `get_output` in the following ways:
+
+```hcl
+inputs = {
+  instance_id_directly = get_output("${get_terragrunt_dir()}/../../app1", "instance_id")
+  instance_name_directly = get_output("${get_terragrunt_dir()}/../../app1", "instance_name")
+  instance_id_indirectly = get_output("${get_terragrunt_dir()}/../../app1")["instance_id"]
+}
+```
+
+Note that the terragrunt run configuration is inherited from the current session. What this means is that settings set
+with environment variables (such as `TERRAGRUNT_IAM_ROLE`) are automatically passed down to the execution context of
+the target terragrunt config when reading the output data.
+
+**Important**: Because `get_output` depends on the target config having been applied already, you will need to make sure
+that the specified `TARGET_TERRAGRUNT_CONFIG` is listed as a dependency in the `dependencies` block. Otherwise, this
+will fail when you run an `apply-all` since the output will be interpolated before the target config has applied.
+Additionally, during an `apply-all`, the terragrunt configuration has to be partially interpolated in order to build up
+the dependency tree. This means that you will have issues using `apply-all` if `get_output` is used in the following
+blocks, as it will be interpolated prior to any modules being applied:
```

Comment:
Not sure I follow this last one? Can you give an example?

Reply:
Imagine you had:

```
live
├── sql
│   └── terragrunt.hcl
└── vpc
    └── terragrunt.hcl
```

And you had:

```
locals = {
  vpc_path = "../vpc"
  vpc_id = get_output(local.vpc_path, "vpc_id")
}

dependencies {
  paths = [local.vpc_path]
}

inputs = {
  vpc_id = local.vpc_id
}
```

Now suppose we have a completely clean project and nothing is applied yet and we run `apply-all`. In order to build the dependency tree, `terragrunt` needs to parse out the `dependencies` block and decode them *before any modules have been applied*. In the above use case, the `dependencies` block also references `locals`, so we need to decode `locals` as well. As part of that, we need to interpolate `get_output`.

Well, `get_output` will try to read the output of the `vpc` module and fail, because that hasn't been applied yet, because we are still in the phase of building up the dependency tree.

Reply:
Oh, I gotcha. So then the `get_output` function would have to:

1. Is this a single-module command (e.g., `apply`) or multi-module (e.g., `apply-all`).
1. If single-module, and the dependency has not yet been applied (i.e., output is missing), return an error immediately. 
1. If multi-module, add the VPC to the dependency graph, pause config parsing until the command (e.g., `apply`) has been executed in that dependency, and then continue config parsing after.

Would that work?

Reply:
This still exists, but is less of an issue because of the way you access `terragrunt_output`.